### ISSUE TITLE
In persistent mode lost Keys.

### ISSUE THREAD

**Issue by hsnamed** [OpenHFT/Chronicle-Map#233]
Steps to reproduce:
```    
@Test
    public void lostkeysTest() throws Exception{
        File storagetmp = new File("./tmp.dat");
        if(storagetmp.exists()) storagetmp.delete();
        storagetmp.deleteOnExit();

        LongValue valueSample = Values.newHeapInstance(LongValue.class);
        ChronicleMap<byte[], LongValue> mmap = ChronicleMap
                .of(byte[].class, LongValue.class)
                .entries(10)
                .actualSegments(1)
                .constantValueSizeBySample(valueSample)
                .constantKeySizeBySample(new byte[36])
                .removeReturnsNull(false)
                .createPersistedTo(storagetmp);

        valueSample.setValue(1);
        var lostKey = "12b5633bad1f9c167d523ad1aa1947b2732a865bf5414eab2f9e5ae5d5c191ba00000001";

        var h1add = Hex.decodeHex("591e91f809d716912ca1d4a9295e70c3e78bab077683f79350f101da6458807300000000");
        mmap.put(h1add, valueSample);

        var bLostKey = Hex.decodeHex(lostKey);
        mmap.put(bLostKey, valueSample);

        var v1 = mmap.remove(h1add);
        Assert.assertEquals(1L, v1.getValue());

        var h3add = Hex.decodeHex("6f80ca7441710cf0942cf99e3e8aa59d38d73b124730306afeb1aec8f08fd76b00000000");
        mmap.put(h3add, valueSample);
        //here we lost key
        var lostKeyVal = mmap.get(bLostKey);
        Assert.assertEquals(1L, lostKeyVal.getValue());

        var v2 = mmap.remove(bLostKey);
        Assert.assertEquals(1L, v2.getValue());
}
 ```
map stat :
```
09:02:36.892 [Test worker] DEBUG ChMapTest - Map  segments: 1, size: 2
09:02:36.893 [Test worker] DEBUG ChMapTest -   segment[0] - 2 entries
09:02:36.894 [Test worker] DEBUG ChMapTest - 	 266d63a1ad1f9c167d523ad1aa1947b2732a865bf5414eab2f9e5ae5d5c191ba00000001, 36 bytes -> LongValue{ value=1 }, 8 bytes
09:02:36.894 [Test worker] DEBUG ChMapTest - 	 6f80ca7441710cf0942cf99e3e8aa59d38d73b124730306afeb1aec8f08fd76b00000000, 36 bytes -> LongValue{ value=1 }, 8 bytes
```
As we see strange key : <266d63a1>ad1f9c167d523ad1aa1947b2732a865bf5414eab2f9e5ae5d5c191ba00000001
its like <12b5633b>ad1f9c167d523ad1aa1947b2732a865bf5414eab2f9e5ae5d5c191ba00000001 except 4 bytes

Problem arise when map used in persisted mode **createPersistedTo**

Compiled with JDK 15 , Runtime JDK 11
```
compile group: 'net.openhft', name: 'chronicle-map', version: '3.20.61'
```

**Comment by hsnamed** [comment]
I've found that depends on number segments , so if i changed number of entries to 100_000_000 
exception about key not found arise earlier than before/

**Comment by hsnamed** [comment]
Code to reproduce :
hashes : [https://pastebin.com/sKpRDgZE](https://pastebin.com/sKpRDgZE)

```
public class ChMapTest implements Loggable {

    File storagetmp;
    long maxNb;

    LongValue valueSample = Values.newHeapInstance(LongValue.class);
    ChronicleMap<byte[], LongValue> mmap;

    @Before
    public void setUp() throws Exception {
        storagetmp = new File("./unspentsdb/tmp.dat");
        if(storagetmp.exists())
            storagetmp.delete();

        maxNb = 1_0;
        var keySample = new byte[36];

        mmap = ChronicleMap
                .of(byte[].class, LongValue.class)
                .name("map1")
                .entries(300)
                .constantValueSizeBySample(valueSample)
                .constantKeySizeBySample(keySample)
                .checksumEntries(true)
                .removeReturnsNull(false)
                .createPersistedTo(storagetmp);
    }


```
    @Test
    public void addManyRemoveOne() throws Exception {
        var in = ChMapTest.class.getClassLoader().getResourceAsStream("hashes");
        BufferedReader reader = new BufferedReader(new InputStreamReader(in));
        while(reader.ready()) {
            String[] line = reader.readLine().split(",");
            var op = line[0];
            var hash = Hex.decodeHex(line[1]);
            if (op.equals("add")) {
                var val = Long.valueOf(line[2]);
                valueSample.setValue(val);
                mmap.put(hash, valueSample);
            }
            if (op.equals("del")) {
                var val = mmap.remove(hash);
                if (val == null) {
                    throw new Exception("key unmapped : " + Hex.encodeHexString(hash));
                }
            }

        }
    }
}

key unmapped : 12b5633bad1f9c167d523ad1aa1947b2732a865bf5414eab2f9e5ae5d5c191ba00000001
java.lang.Exception: key unmapped : 12b5633bad1f9c167d523ad1aa1947b2732a865bf5414eab2f9e5ae5d5c191ba00000001

**Comment by minborg** [comment]
I have added a reproducing test. Once it passes on all platforms, this issue will be closed automatically..

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.22ea6](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.22ea6), [BOM-2.22ea86](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.22ea86)
